[quant][refactor tests] Move qtensor serialization tests from test_deprecated_jit

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59089 [quant][refactor tests] Move qtensor serialization tests from test_deprecated_jit**
* #59088 [quant][refactor tests] Clean up test_quantization.py

Summary:
Move these tests into test_quantized_tensor

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28750065](https://our.internmc.facebook.com/intern/diff/D28750065)

supriyar committed:
[quant][refactor tests] Move qtensor serialization tests from test_deprecated_jit
Summary:
Move these tests into test_quantized_tensor

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28750065).

supriyar committed:
Update on "[quant][refactor tests] Move qtensor serialization tests from test_deprecated_jit"


Summary:
Move these tests into test_quantized_tensor

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28750065](https://our.internmc.facebook.com/intern/diff/D28750065)

[ghstack-poisoned]

jerryzh168 commented on test/quantization/core/test_quantized_tensor.py:
```diff
@@ -903,3 +903,65 @@ def test_choose_qparams_optimized(self):
             self.assertEqual(y[0].numpy(), ref[0])
             self.assertEqual(y[1].numpy(), ref[1])
+
+class TestQTensorSerialization(TestCase):
```
shall we just merge this with TestQuantizedTensor?

supriyar replied:
sure

jerryzh168 commented on test/quantization/core/test_quantized_tensor.py:
```diff
@@ -903,3 +903,65 @@ def test_choose_qparams_optimized(self):
+    def test_pickle_checkpoint_qtensor(self):
+        self._test_pickle_checkpoint_qtensor('cpu')
+
+    def test_serialize_qtensor(self):
```
nit: we can rename this to test_jit_serialization since it's specific to torch.jit.save and torch.jit.load

supriyar committed:
Update on "[quant][refactor tests] Move qtensor serialization tests from test_deprecated_jit"


Summary:
Move these tests into test_quantized_tensor

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28750065](https://our.internmc.facebook.com/intern/diff/D28750065)

[ghstack-poisoned]

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28750065).
